src/averageAges.js

```diff
@@ -38,6 +58,29 @@ function calculateMenAverageAge(people, century) {
  */
 function calculateWomenAverageAge(people, withChildren) {
   // write code here
+  if (!withChildren) {
+    const arrOfWomen = [...people].filter((person) =>
+      person.sex === 'f');
+
+    const ages = arrOfWomen.map((person) =>
+      person.died - person.born);
+
+    return ages.reduce((accumulator, age) =>
+      (accumulator + age
+      ), 0) / ages.length;
+  }
+
+  const arrOfWomenWithChildren = [...people].filter((person) =>
+    person.sex === 'f' && people.find((person1) =>
+      person.name === person1.mother));
+
+  const agesOfWomenWithChildren = arrOfWomenWithChildren.map((person) =>
+    person.died - person.born);
+
+  const result = agesOfWomenWithChildren.reduce((accumulator, age) =>
+    (accumulator + age), 0) / agesOfWomenWithChildren.length;
```

Comment:
You do the same thing heare as in the previous function, create a function-helper for it